coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallTransformers.java

```diff
@@ -0,0 +1,206 @@
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.UserAgentParser", 2, "useragentparser",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.lookup.hive.BrowserLookup", 3, "browserlookup",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.jobs.udf.hive.ConvertIndustryCode", 1, "converttoindustryv1",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper", 1,
+        "urn_extractor", linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper", 1, "urn_extractor",
+        linkedInFunctionSignatureSet);
+
+    addLinkedInFunctionTransformerFromHiveRegistry(DEFAULT_SQL_CALL_TRANSFORMER_LIST, linkedInFunctionSignatureSet);
+  }
+
+  private static void addLinkedInFunctionTransformer(String linkedInFuncName, int numOperands, String trinoFuncName,
+      Set<String> linkedInFunctionSignatureSet) {
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(new OperatorBasedSqlCallTransformer(
+        linkedInFunctionToCoralSqlOperator(linkedInFuncName), numOperands, trinoFuncName));
+    linkedInFunctionSignatureSet.add(linkedInFuncName + "_" + numOperands);
+  }
+
+  private static void addAdHocTransformers() {
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(new MapStructAccessOperatorTransformer());
+  }
+
+  private static SqlOperator linkedInFunctionToCoralSqlOperator(String className) {
+    return HIVE_FUNCTION_REGISTRY.lookup(className).iterator().next().getSqlOperator();
+  }
+
+  private static void addLinkedInFunctionTransformerFromHiveRegistry(List<SqlCallTransformer> sqlCallTransformerList,
+      Set<String> linkedInFunctionSignatureSet) {
+    ImmutableMultimap<String, Function> registry = HIVE_FUNCTION_REGISTRY.getRegistry();
+    Converter<String, String> caseConverter = CaseFormat.UPPER_CAMEL.converterTo(CaseFormat.LOWER_UNDERSCORE);
+    for (Map.Entry<String, Function> entry : registry.entries()) {
+      // we cannot use entry.getKey() as function name directly, because keys are all lowercase, which will
+      // fail to be converted to lowercase with underscore correctly
+      final String hiveFunctionName = entry.getValue().getFunctionName();
+      if (!hiveFunctionName.startsWith("com.linkedin")) {
+        continue;
+      }
+      String[] nameSplit = hiveFunctionName.split("\\.");
+      // filter above guarantees we've at least 2 entries
+      String className = nameSplit[nameSplit.length - 1];
+      String funcName = caseConverter.convert(className);
+      SqlOperator op = entry.getValue().getSqlOperator();
+      for (int i = op.getOperandCountRange().getMin(); i <= op.getOperandCountRange().getMax(); i++) {
+        if (!linkedInFunctionSignatureSet.contains(hiveFunctionName.toLowerCase() + "_" + i)) {
```

Comment:
If you use `hiveFunctionName.toLowerCase()` here, the above `linkedInFunctionSignatureSet.add(linkedInFuncName + "_" + numOperands);` should be `linkedInFunctionSignatureSet.add(linkedInFuncName.toLowerCase() + "_" + numOperands);`

Reply:
Done.